src/main/perf/Indexer.java

```diff
@@ -205,6 +205,15 @@ private static void _main(String[] clArgs) throws Exception {
     } 
 
     final String lineFile = args.getString("-lineDocsFile");
+    String vectorFile;
+    int vectorDimension;
+    if (args.hasArg("-vectorFile")) {
+        vectorFile = args.getString("-vectorFile");
+        vectorDimension = args.getInt("-vectorDimension");
```

Comment:
Do we check, while actually reading the vectors, that the specified dimension matches what we see in the file?

Or, I guess Lucene will do this check anyways for us, so that's good enough.

Reply:
Well, the file has no metadata, so there is no way of checking. Possibly I could add a header with the dimension? Then we wouldn't have to pass it around as an argument everywhere.

Reply:
Let's not block pushing for this!  This PR looks like a great step forward already!